### ISSUE TITLE
SSL Configuration - Keystore path change doesn't work

### ISSUE THREAD

**Issue by kandrej** [eclipse/kura#1849]
When I try to change the keystore path the property is saved, but the default java keystore is used.
There is a warning inside kura.log saying "null old password"

**Comment by MMaiero** [comment]
Hi,
which version of Kura are you using?
What are the exact steps you did to perform the operation?

I've just done the change of keystore from /opt/eclipse/kura/security/cacerts.ks (the referenced file did not exist) to /opt/eclipse/security/cacerts.ks, setting also the proper password for the new keystore and this is the log I get in kura.log file:

```
2017-12-22 15:15:38,154 [qtp462890-78] WARN  o.e.k.n.a.v.l.WpaSupplicantConfigReader - WPA in client mode is not configured
2017-12-22 15:15:38,203 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Loading init configurations from: 1513952753472...
2017-12-22 15:15:38,365 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Merging configuration for pid: org.eclipse.kura.ssl.SslManagerService
2017-12-22 15:15:38,393 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Updating Configuration of ConfigurableComponent org.eclipse.kura.ssl.SslManagerService ... Done.
2017-12-22 15:15:38,399 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.s.SslManagerServiceImpl - updated...
2017-12-22 15:15:38,401 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.d.DataServiceImpl - Notified DataTransportService configuration updating...
2017-12-22 15:15:38,401 [Component Resolve Thread (Bundle 7)] WARN  o.e.k.c.d.t.m.MqttDataTransport - MQTT client already disconnected
2017-12-22 15:15:38,401 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.d.t.m.MqttDataTransport - Building new configuration...
2017-12-22 15:15:38,518 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.d.DataServiceImpl - Notified DataTransportService configuration updated.
2017-12-22 15:15:38,745 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Writing snapshot - Saving /opt/eclipse/kura/data/snapshots/snapshot_1513955738394.xml...
2017-12-22 15:15:38,812 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Writing snapshot - Saving /opt/eclipse/kura/data/snapshots/snapshot_1513955738394.xml... Done.
2017-12-22 15:15:38,814 [qtp462890-78] INFO  o.e.k.c.c.ConfigurationServiceImpl - Snapshots Garbage Collector. Deleting /opt/eclipse/kura/data/snapshots/snapshot_1513949940487.xml
```

**Comment by kandrej** [comment]
Kura version is 3.1.0
The step are:
1. Create a keystore at the default location /opt/eclipse/kura/security/cacerts.ks
2. In SSL Configuration change the password for this keystore
3. Create a new keystore. Let's say /opt/eclipse/kura/security/cacerts2.ks
4. In SSL Configuration change the keystore path to /opt/eclipse/kura/security/cacerts2.ks and set the new password


2017-12-22 19:56:41,547 [qtp9304820-36] INFO  o.e.k.c.c.ConfigurationServiceImpl - Merging configuration for pid: org.eclipse.kura.ssl.SslManagerService
2017-12-22 19:56:41,580 [qtp9304820-36] INFO  o.e.k.c.c.ConfigurationServiceImpl - Updating Configuration of ConfigurableComponent org.eclipse.kura.ssl.SslManage
rService ... Done.
2017-12-22 19:56:41,587 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.s.SslManagerServiceImpl - updated...
2017-12-22 19:56:41,588 [Component Resolve Thread (Bundle 7)] WARN  o.e.k.c.s.SslManagerServiceImpl - null old password
2017-12-22 19:56:41,589 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.d.DataServiceImpl - Notified DataTransportService configuration updating...
2017-12-22 19:56:41,589 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.c.d.t.m.MqttDataTransport - Disconnecting...

**Comment by MMaiero** [comment]
Closing as it should be solved now after the merge of PR #2095

**Comment by Amar82** [comment]
Create a keystore at the default location /opt/eclipse/kura/security/cacerts.ks 

How to do this.. in my Kura , the file or folder does not exist...
